doc/sdk-cli-reference.md

```diff
@@ -0,0 +1,177 @@
+ ---> Running in 34b4bb507c14
+Removing intermediate container 34b4bb507c14
+ ---> c671ec1cff03
+Step 4/4 : USER example-operator
+ ---> Running in bd336926317c
+Removing intermediate container bd336926317c
+ ---> d6b58a0fcb8c
+Successfully built d6b58a0fcb8c
+Successfully tagged quay.io/example/operator:v0.0.1
+```
+
+## completion
+
+### Available Commands
+
+#### bash - Generate bash completions
+
+##### Flags
+
+* `-h, --help` - help for bash
+
+#### zsh - Generate zsh completions
+
+##### Flags
+
+* `-h, --help` - help for zsh
+
+### Flags
+
+* `-h, --help` - help for completion
+
+### Use
+
+Generators for shell completions
+
+Example:
+
+```bash
+operator-sdk completion bash
+
+// Output:
+# bash completion for operator-sdk                         -*- shell-script -*-
+...
+# ex: ts=4 sw=4 et filetype=sh
+```
+
+## generate
+
+### Available Commands
+
+#### k8s - Generates Kubernetes code for custom resource
+
+##### Use
+
+k8s generator generates code for custom resource given the API spec
+to comply with kube-API requirements.
+
+##### Flags
+
+* `-h, --help` - help for k8s
+
+#### olm-catalog - Generates OLM Catalog manifests
+
+##### Flags
+
+* `--image` **(required)** string - The container image name to set in the CSV to deploy the operator e.g: quay.io/example/operator:v0.0.1
+* `--version` **(required)** string - The version of the current CSV e.g: 0.0.1
+* `-h, --help` - help for olm-catalog
+
+##### Example
+
+`operator-sdk generate olm-catalog --image=quay.io/example/operator:v0.0.1 --version=0.0.1`
+
+### Flags
+
+* `-h, --help` - help for generate
+
+## new
+
+The operator-sdk new command creates a new operator application and
+generates a default directory layout based on the input `project-name`.
+
+### Args
+
+* `project-name` - the project name of the new 
+
+### Flags
+
+* `--api-version` **(required)** string - Kubernetes apiVersion and has a format of `$GROUP_NAME/$VERSION` (e.g app.example.com/v1alpha1)
+* `--kind` **(required)** string - Kubernetes CustomResourceDefintion kind. (e.g AppService)
+* `-h, --help` - help for new
+
+### Example
+
+```bash
+mkdir $GOPATH/src/github.com/example.com/
+cd $GOPATH/src/github.com/example.com/
+operator-sdk new app-operator --api-version=app.example.com/v1alpha1 --kind=AppService
+
+//Output
+Create app-operator/.gitignore
+...
+
+```
+
+## up
+
+### Available Commands
+
+#### local - Launches the operator locally
+
+##### Use
+
+The operator-sdk up local command launches the operator on the local machine
+by building the operator binary with the ability to access a
+kubernetes cluster using a kubeconfig file.
+
+##### Flags
+
+* `--kubeconfig` string - The file path to kubernetes configuration file; defaults to $HOME/.kube/config
+
+* `--namespace` string - The namespace where the operator watches for changes. (default "default")
+
+* `--operator-flags` - Flags that the local operator may need.
+
+* `-h, --help` - help for local
+
+##### Example
```

Comment:
I'll let @shawn-hurley to provide additional descriptions on how to use those flags as he worked on this in a future PR.